src/net/SCTP/small/BlockingSCTPStream.cs

```diff
@@ -32,14 +32,25 @@ namespace SIPSorcery.Net.Sctp
     public class BlockingSCTPStream : SCTPStream
     {
         private ConcurrentDictionary<int, SCTPMessage> undeliveredOutboundMessages = new ConcurrentDictionary<int, SCTPMessage>();
-
         private static ILogger logger = Log.Logger;
+        private SemaphoreSlim semaphore = new SemaphoreSlim(1);
 
         public BlockingSCTPStream(Association a, int id) : base(a, id) { }
 
         public override void send(string message)
         {
-            lock (this)
+            sendasync(message).GetAwaiter().GetResult();
+        }
+
+        public override void send(byte[] message)
+        {
+            sendasync(message).GetAwaiter().GetResult();
+        }
+
+        public override async Task sendasync(byte[] message)
+        {
+            await semaphore.WaitAsync().ConfigureAwait(false);

```

Comment:
Is this the lock you're referring to in your comment? I'd tend to agree that since undeliveredOutboundMessages is concurrent it's not needed.